src/User/Auth/authService/authService.js

```diff
@@ -0,0 +1,66 @@
+import AuthUser from "../AuthUser.js"
+import bcrypt from "bcrypt";
+import { v4 as uuidv4 } from 'uuid';
+import mailService from "./mailService.js";
+import Post from "../../../Post/Post.js";
+
+class AuthService {
+    async registration(username, password, email) {
+        const newUser = await AuthUser.findOne({ username })
+        const newEmail = await AuthUser.findOne({ email })
+        if (newUser || newEmail) {
+            throw new Error('such user exists')
+        }
+        const hashPassword = await bcrypt.hash(password, 5)
+        const confirmationCode = uuidv4();
+        try {
+            const user = await AuthUser.create({ username, password: hashPassword, email, confirmationCode })
+            await mailService.sendActivationMail(username, email, confirmationCode)
+            return user
+        }
+        catch (e) {
+            console.log(e)
+        }
+
+    }
+    async activate(confirmationCode) {
+        const user = await AuthUser.findOne({ confirmationCode })
```

Comment:
Seems we need to add linter to check code styles. There are no ; in the end of string

Reply:
added linter